lib/services/elasticsearch.js

```diff
@@ -874,7 +879,28 @@ class ElasticSearch extends Service {
   }
 
   /**
-   * Adds a mapping definition to a specific type
+   * Updates a collection mappings and settings
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} body - mappings ({}), settings ({})
+   *
+   * @return {Promise}
+   */
+  async updateCollection (index, collection, { mappings={}, settings={}} = {}) {
```

Comment:
The index is left in an inconsistent state (from a user's viewpoint) if applying the mapping succeeds, but applying the settings fail.

I didn't see an easy way to play these 2 operations as 1 in ES, nor a way to easily rollback either of them.

Since mappings cannot be rolled back at all, maybe we could instead get the current settings, apply the new ones, then apply the mappings, and if that fails, re-apply the previous settings before throwing an error.

What do you think?

Reply:
Yes you're right, I'm gonna change this